samples/bluetooth/iso_connected_benchmark/src/main.c

```diff
@@ -0,0 +1,1042 @@
+	}
+
+	LOG_INF("Binding ISO");
+	err = bt_iso_chan_bind(conn_pointers, cis_create_count, chan_pointers);
+	if (err != 0) {
+		LOG_ERR("Failed to bind iso to connection: %d", err);
+		return err;
+	}
+
+	LOG_INF("Connecting ISO channels");
+	err = bt_iso_chan_connect(chan_pointers, cis_create_count);
+	if (err != 0) {
+		LOG_ERR("Failed to connect iso: %d", err);
+		return err;
+	}
+	total_iso_conn_count++;
+
+	for (int i = 0; i < cis_create_count; i++) {
+		err = k_sem_take(&sem_iso_connected, K_FOREVER);
+		if (err != 0) {
+			LOG_ERR("failed to take sem_iso_connected: %d", err);
+			return err;
+		}
+	}
+
+	return 0;
+}
+
+static void reset_sems(void)
+{
+	(void)k_sem_reset(&sem_adv);
+	(void)k_sem_reset(&sem_iso_accept);
+	(void)k_sem_reset(&sem_iso_connected);
+	(void)k_sem_reset(&sem_iso_disconnected);
+	(void)k_sem_reset(&sem_connected);
+	(void)k_sem_reset(&sem_disconnected);
+}
+
+static int test_run_central(void)
+{
+	int err;
+	char c;
+
+	iso_conn_start_time = 0;
+	last_received_counter = 0;
+	memset(&stats_current_conn, 0, sizeof(stats_current_conn));
+	reset_sems();
+
+	printk("Change ISO settings (y/N)?\n");
+	c = console_getchar();
+	if (c == 'y' || c == 'Y') {
+		err = change_central_settings();
+		if (err != 0) {
+			LOG_ERR("Failed to set parameters: %d", err);
+			return err;
+		}
+	}
+
+	err = central_create_connection();
+	if (err != 0) {
+		LOG_ERR("Failed to create connection: %d", err);
+		return err;
+	}
+
+	err = central_create_cis();
+	if (err != 0) {
+		LOG_ERR("Failed to create connection: %d", err);
+		bt_conn_unref(default_conn);
+		return err;
+	}
+
+	k_work_init_delayable(&iso_send_work, iso_timer_timeout);
+	k_work_schedule(&iso_send_work, K_MSEC(0));
+
+	err = k_sem_take(&sem_disconnected, K_FOREVER);
+	if (err != 0) {
+		LOG_ERR("failed to take sem_disconnected: %d", err);
+		bt_conn_unref(default_conn);
+		return err;
+	}
+
+	for (int i = 0; i < cis_create_count; i++) {
+		err = k_sem_take(&sem_iso_disconnected, K_FOREVER);
+		if (err != 0) {
+			LOG_ERR("failed to take sem_iso_disconnected: %d", err);
+			bt_conn_unref(default_conn);
+			return err;
+		}
+	}
+
+	LOG_INF("Disconnected - Cleaning up");
+	bt_conn_unref(default_conn);
+	k_work_cancel_delayable(&iso_send_work);
+
+	return 0;
+}
+
+static int peripheral_cleanup(void)
+{
+	int err;
+
+	err = k_sem_take(&sem_disconnected, K_NO_WAIT);
+
+	if (err != 0) {
+		err = bt_conn_disconnect(default_conn, BT_HCI_ERR_REMOTE_USER_TERM_CONN);
+		if (err != 0) {
+			LOG_ERR("Could not disconnect ACL: %d", err);
+			return err;
+		}
+	} /* else ACL already disconnected */
+
+	for (int i = 0; i < cis_create_count; i++) {
+		err = k_sem_take(&sem_iso_disconnected, K_NO_WAIT);
+		if (err != 0) {
+			err = bt_iso_chan_disconnect(&iso_chans[i]);
+			if (err != 0) {
+				LOG_ERR("Could not disconnect ISO[%d]: %d", i, err);
+				break;
+			}
+		} /* else ISO already disconnected */
+	}
+
+	return err;
+}
+
+static int test_run_peripheral(void)
```

Comment:
nit: the name of the function looks like one from unit tests. Maybe change to run_benchmark_for_peripheral?

Reply:
See https://github.com/zephyrproject-rtos/zephyr/pull/36752#discussion_r668525632